kensu/utils/kensu.py

```diff
@@ -572,3 +640,62 @@ def new_lineage(self, process_lineage_dependencies, report_stats=True, **kwargs)
+                      stats=stats,
+                      extra_as_json=None)._report()
+
+    def send_rules(self):
+        if self.rules and self.api_url:
+            process_id = self.process.to_guid()
+            project_id = self.process_run.projects_refs[0].by_guid
+            env_name = self.process_run.environment
+            cv = self.process_run.executed_code_version_ref.by_guid
+
+            for map in self.rules:
+                for lds_id in map:
+                    field_name = map[lds_id]['field']
+                    fun = map[lds_id]['fun']
+
+                    data = self.sdk.get_lineages_in_project(project_id, process_id, env_name, cv)
+
+                    lds_guid = [e['datasource'] for e in data['data']['nodes'] if e['datasource']['name'] == lds_id][0][
+                        'id']
+                    lineage_ids = set(
+                        [data['data']['links'][i]['lineage']['id'] for i in range(len(data['data']['links']))])
+                    for lineage_id in lineage_ids:
+                        current_rules = self.sdk.get_rules_for_ds(lds_guid, lineage_id, project_id, env_name)
+                        current_range_rules = {i['fieldName']: i['uuid'] for i in current_rules['data']['predicates'] if
+                                               i['functionName'] == 'Range' and (i['environment'] == env_name)}
+                        current_frequency_rule = [i['uuid'] for i in current_rules['data']['predicates'] if
+                                                  i['functionName'] == 'Frequency' and (i['environment'] == env_name)]
+                        if fun['name'] == 'Range' and (field_name in current_range_rules):
+                            self.sdk.update_rule(current_range_rules[field_name], fun)
+                        elif fun['name'] == 'Frequency' and current_frequency_rule:
+                            self.sdk.update_rule(current_frequency_rule[0], fun)
+
+                        else:
+                            self.sdk.create_rule(lds_guid, lineage_id, project_id, process_id, env_name, field_name, fun)
+
+
+    def check_local_rules(self):
+        from kensu.utils.exceptions import NrowsConsistencyError
+        for check in self.check_rules:
+            if 'check_nrows_consistency' in check:
+                for datasource in check['check_nrows_consistency']:
+                    for rule in check['check_nrows_consistency'][datasource]:
+                        for field in rule:
+                            if rule[field]['output_nrows'] != rule[field]['input_nrows'] :
+                                e = NrowsConsistencyError(datasource, rule[field]['input_nrows'],
+                                                                rule[field]['output_nrows'])
+                                if self.raise_on_check_failure:
+                                    raise e
+                                else:
+                                    logging.warning(e)
```

Comment:
might want to control/choose the logging level,  according to config, e.g. warning/error - add a helper fn I guess